iohk-monitoring/src/Cardano/BM/Data/MonitoringEval.lhs

```diff
@@ -112,8 +112,25 @@ instance Show MEvExpr where
 If evaluation of a monitoring expression is |True|, then a set of actions are
 executed for alerting.
 \begin{code}
-type MEvAction = Text
+data MEvAction = CreateMessage Severity Text
+               | SetGlobalMinimalSeverity Severity
+               | AlterSeverity LoggerName Severity
+               deriving (Eq)
+
+instance FromJSON MEvAction where
+    parseJSON (String s) =
+        case parseEitherAction s of
```

Comment:
We can use directly https://hackage.haskell.org/package/aeson-1.4.3.0/docs/Data-Aeson-Types.html#v:parseEither